classes/mail/variables/Variable.php

```diff
@@ -15,8 +15,36 @@
 
 namespace PKP\mail\variables;
 
+use Illuminate\Support\Arr;
+use PKP\context\Context;
+use PKP\mail\Mailable;
+use Exception;
+
 abstract class Variable
 {
+    protected Mailable $mailable;
+
+    public function __construct(Mailable $mailable)
+    {
+        $this->mailable = $mailable;
+    }
+
+    /**
+     * Retrieve mailable context from associated variables, see pkp/pkp-lib#8204
+     */
+    protected function getContextFromVariables(): Context
```

Comment:
What about calling this `getContext()` since we're already working within variables. Or maybe `getEmailContext()` or `getMailableContext()`? The developer doesn't need to know precisely where it comes from, just what they're getting.